core/src/main/java/com/linecorp/armeria/client/logging/LoggingClient.java

```diff
@@ -83,6 +84,8 @@
     private final Function<Object, Object> requestContentSanitizer;
     private final Function<HttpHeaders, HttpHeaders> responseHeadersSanitizer;
     private final Function<Object, Object> responseContentSanitizer;
+    private final Function<ClientRequestContext, ContentPreviewWriter> requestContentPreviewWriterMapper;
+    private final Function<ClientRequestContext, ContentPreviewWriter> responseContentPreviewWriterMapper;
```

Comment:
Content previewer logic should be independent from any decorators, because the preview should be accessible without any decorators. Could you move the previewer settings to somewhere else, like `AbstractVirtualHostBuilder` or `ServerBuilder`?